scripts/perftune.py

```diff
@@ -649,6 +649,27 @@ def __intel_irq_to_queue_idx(self, irq):
         else:
             return sys.maxsize
 
+    def __mlx_irq_to_queue_idx(self, irq):
+        """
+        Return the HW queue index for a given IRQ for Mellanox NICs in order to sort the IRQs' list by this index.
+
+        Mellanox NICs have the IRQ which name looks like
+        this:
+        mlx5_comp23
+             mlx5_comp<index>
+        or this:
+        mlx4-6
+             mlx4-<index>
+
+        :param irq: IRQ number
+        :return: HW queue index for Mellanox NICs and 0 for all other NICs
+        """
+        mlx_fp_irq_re = re.compile("mlx5_comp(\d+)|mlx4\-(\d+)")
+
+        m = mlx_fp_irq_re.search(self.__irqs2procline[irq])
```

Comment:
Not good. What if search has failed?
See what I do in `__intel_irq_to_queue_idx` for that case.

Reply:
fixed in the same commit, added comment:
comparator for mlx irqs should detect if mlx is not present